### PR TITLE
Convert express-server task to async await

### PR DESCRIPTION
**Pull request by locks**

### CODE REVIEW

File: lib/tasks/server/express-server.js [hunk 1]
```diff
@@ -153,29 +153,30 @@ class ExpressServerTask extends Task {
...
-          this.ui.writeLine('');
-        })
-        .catch((err) => {
-          this.ui.writeError(err);
-        })
-        .finally(() => {
-          this.serverRestartPromise = null;
-        });
+      try {
+        this.serverRestartPromise = await this.stopHttpServer()
```
Comment (Turbo87) [inline lib/tasks/server/express-server.js:159]: I'm not sure about the `await` here. `serverRestartPromise` will no longer contain the promise in this case, but instead the resolution value of the promise 🤔

Comment (locks) [reply lib/tasks/server/express-server.js]: Good point! I think with the new commit the correct logic is preserved.

File: lib/tasks/server/express-server.js [hunk 2]
```diff
@@ -153,29 +153,30 @@ class ExpressServerTask extends Task {
     this.scheduleServerRestart();
   }
 
-  restartHttpServer() {
+  async restartHttpServer() {
     if (!this.serverRestartPromise) {
-      this.serverRestartPromise = this.stopHttpServer()
-        .then(() => {
+      try {
```
Comment (rwjblue) [inline lib/tasks/server/express-server.js:158]: This seems wrong to me, the `try`/`catch`/`finally` should be around the async just below now the creation of the `this.serverRestartPromise` chain.

You need something like this:

```js
this.serverRestartPromise = (async () => {
  try {
    await this.stopHttpServer();

    this.emit('restart');
    this.ui.writeLine('');
    this.ui.writeLine(chalk.green('Server restarted.'));
    this.ui.writeLine('');
  } catch(err) {
    this.ui.writeError(err);
  } finally {
    this.serverRestartPromise = null;
  }
})();
```

Comment (locks) [reply lib/tasks/server/express-server.js:162]: Updated!

**Pull request merged**